corehq/form_processor/migrations/0094_add_partial_index_xforms.py

```diff
@@ -0,0 +1,17 @@
+# Generated by Django 3.2.20 on 2023-09-07 18:48
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('form_processor', '0093_rename_models'),
+    ]
+
+    operations = [
+        migrations.AddIndex(
+            model_name='xforminstance',
+            index=models.Index(condition=models.Q(('deleted_on__isnull', False)), fields=['deleted_on'], name='form_processor_xf_a66494ec_idx'),
+        ),
```

Comment:
Just thinking about this some more - won't this lock the table while the index is being created?

You may need to take a more manual approach to creating the index so that you can make it a 'concurrent' operation. Here's an example: https://github.com/dimagi/commcare-hq/blob/0bae5ccf834ce0bc97d3e36470a05f4049665b0a/corehq/apps/sms/migrations/0045_auto_20200902_0938.py#L15

It's probably also a good idea to split the index addition into a separate PR so that it can be rolled out independent of the other changes (since it may require running outside of a deploy - at least on prod).

Reply:
thanks for the example Simon! and good call in separating these out... I probably should've realized this needed to be two PRs when I put that '+' in the PR title